### PR TITLE
ipptool: support mdns uris

### PR DESCRIPTION
**Pull request by zdohnal**
See https://github.com/apple/cups/pull/5793

### CODE REVIEW

File: cups/ipp-vars.c [hunk 1]
```diff
@@ -221,8 +222,29 @@ _ippVarsSet(_ipp_vars_t *v,		/* I - IPP variables */
   if (!strcmp(name, "uri"))
   {
     char		uri[1024];	/* New printer URI */
+    char		resolved[1024];	/* Resolved mDNS URI */
+    char		value_uri[1024];	/* URI from value */
+    http_uri_status_t	uri_status;	/* URI status */
 
-    if (httpSeparateURI(HTTP_URI_CODING_ALL, value, v->scheme, sizeof(v->scheme), v->username, sizeof(v->username), v->host, sizeof(v->host), &(v->port), v->resource, sizeof(v->resource)) < HTTP_URI_STATUS_OK)
+    snprintf(value_uri, sizeof(value_uri), "%s", value);
```
Comment (michaelrsweet) [inline cups/ipp-vars.c:229]: OK, so this copy is 100% unnecessary. Just call strstr on the original value string and then reassign value to the resolved URI.

Also, for future reference snprintf correctly adds a nul terminator at the end - you don't need to treat it like strncpy. Stylistically you should use "sizeof(foo) - 1" and not hardcode the last element, and for CUPS code you can use strlcpy (emulated when needed) which is a safe replacement for strncpy that makes sure the nul is on the end of the buffer. snprintf should only be used if you actually need to add something to the string.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
See comment attached to the review; I don't see a reason not to support this, we just need to remove the extra string copies.

File: cups/ipp-vars.c (hunk 1 above)
Comment (zdohnal) [reply cups/ipp-vars.c:229]: You're right, I'll update the code. Thank you for the tips too!

**Comment by zdohnal** [comment]
The code is updated, feel free to review.

**Review by michaelrsweet — APPROVED** [review]
+1

There is some minor stylistic cleanup (no need for <cups/cups.h> or the braces around the return) but I like the new changes!

**Comment by zdohnal** [comment]
@michaelrsweet (facepalm) I'm sorry, I missed those two (didn't check cups-private.h contains cups/cups.h too, so no need to include it separately/ and those unnecessary {} brackets).

I would like to ask about spaces/tabs in CUPS code - I see the DEVELOPING.md states 2 spaces must be used for indentation in most cases, but there are many tabs used too. What is the CUPS code style regarding those? Tabs or spaces?

IMHO spaces are better for consistent reading experience across any text editor, because they don't be influenced by personal settings like tabs.

**Comment by michaelrsweet** [comment]
@zdohnal The coding style guide allows tabs at 8 columns, and comments on arguments and variables and types typically start at column 41 (5 tabs in). But it doesn’t matter whether you actually use tabs, just that the indentation is correct.

**Pull request merged**